etl/steps/grapher/worldbank_wdi/2022-05-26/wdi.py

```diff
@@ -0,0 +1,304 @@
+            description=create_description(var),
+            sources=[source],
+            unit=unit,
+            short_unit=short_unit,
+            display=display,
+            additional_info=None
+            # licenses=[var['license_type']]
+        )
+
+    if not all([len(table[var_code].sources) == 1 for var_code in var_codes]):
+        missing = [
+            var_code for var_code in var_codes if len(table[var_code].sources) != 1
+        ]
+        raise RuntimeError(
+            "Expected each variable code to have one source, but the following variables "
+            f"do not: {missing}. Are the source names for these variables "
+            "missing from `wdi.sources.json`?"
+        )
+
+    table.reset_index(inplace=True)
+    table["entity_id"] = gh.country_to_entity_id(table["country"], create_entities=True)
+    table = table.drop(columns=["country"]).set_index(["year", "entity_id"])
+
+    for col in table.columns:
+        tb = table[[col]].dropna()
+        if tb.shape[0]:
+            yield tb  # type: ignore
+
+
+def load_clean_source_mapping() -> dict:
+    with open(Path(__file__).parent / "wdi.sources.json", "r") as f:
+        sources = json.load(f)
+        source_mapping = {source["rawName"]: source for source in sources}
+        assert len(sources) == len(source_mapping)
+    return source_mapping
+
+
+def create_description(var: dict) -> Optional[str]:
+    desc = ""
+    if pd.notnull(var["long_definition"]) and len(var["long_definition"].strip()) > 0:
+        desc += var["long_definition"]
+    elif (
+        pd.notnull(var["short_definition"]) and len(var["short_definition"].strip()) > 0
+    ):
+        desc += var["short_definition"]
+
+    if (
+        pd.notnull(var["limitations_and_exceptions"])
+        and len(var["limitations_and_exceptions"].strip()) > 0
+    ):
+        desc += f'\n\nLimitations and exceptions: {var["limitations_and_exceptions"]}'
+
+    if (
+        pd.notnull(var["statistical_concept_and_methodology"])
+        and len(var["statistical_concept_and_methodology"].strip()) > 0
+    ):
+        desc += f'\n\nStatistical concept and methodology: {var["statistical_concept_and_methodology"]}'
+
+    # retrieves additional source info, if it exists.
+    if (
+        pd.notnull(var["notes_from_original_source"])
+        and len(var["notes_from_original_source"].strip()) > 0
+    ):
+        desc += f'\n\nNotes from original source: {var["notes_from_original_source"]}'
+
+    desc = re.sub(r" *(\n+) *", r"\1", re.sub(r"[ \t]+", " ", desc)).strip()
+
+    if len(desc) == 0:
+        desc = None
+
+    return desc
+
+
+def replace_years(s: str, year: Union[int, str]) -> str:
+    """replaces all years in string with {year}.
+
+    Example:
+
+        >>> replace_years("GDP (constant 2010 US$)", 2015)
+        "GDP (constant 2015 US$)"
+    """
+    year_regex = re.compile(r"\b([1-2]\d{3})\b")
+    s_new = year_regex.sub(str(year), s)
+    return s_new
+
+
+class VariableMatcher:
```

Comment:
This is really cool, I can see it being used for other use cases too (just saying, you don't have to do anything now).